infrastructures/docker/src/main/java/org/eclipse/che/workspace/infrastructure/docker/DockerInternalRuntime.java

```diff
@@ -175,57 +175,71 @@ private DockerInternalRuntime(DockerRuntimeContext context,
     @Override
     protected void internalStart(Map<String, String> startOptions) throws InfrastructureException {
         startSynchronizer.setStartThread();
+        try {
+            doStart(startOptions);
+            startSynchronizer.complete(null);
```

Comment:
If interruption happens right before execution of this line we will report that workspace is stopped but destroy of environment won't be called